### PR TITLE
Ability to set custom headers (whether or not headers exist in the file)

### PR DESCRIPTION
**Pull request by kitbs**
If you read a CSV file which does not contain a header row, you will use `noHeaderRow()`, which returns each data row with integer keys.

For example, this file:

```csv
john@example.com,john,doe
mary-jane@example.com,mary jane,doe
```
will be returned as:
```php
[
    [
        0 => 'john@example.com',
        1 => 'john',
        2 => 'doe',
    ],
    [
        0 => 'mary-jane@example.com',
        1 => 'mary jane',
        2 => 'doe',
    ],
]
```

However, if you already know what the headers should be for the file (i.e. they are defined externally), it would be nice to be able to get each data row with the correct headers.

For example, you know the headers should be `email`, `first_name` and `last_name`, so you want the above file to be returned as:
```php
[
    [
        'email' => 'john@example.com',
        'first_name' => 'john',
        'last_name' => 'doe',
    ],
    [
        'email' => 'mary-jane@example.com',
        'first_name' => 'mary jane',
        'last_name' => 'doe',
    ],
]
```

In order to do this currently, you would need to reimplement all the array_slice/array_combine logic from `getValueFromRow()` in your own code.

To support this, I have added a `setHeaders(array $headers)` method, where you can specify the custom headers to use. It affects the output of `getValueFromRow()` and therefore `getRows()`, so that the custom headers are used instead of integer keys.

It can either be used with `noHeaderRow()` (custom headers will be used instead of integer keys), or without (custom headers will be used instead of the headers defined in the file, and the header row will not be returned as a data row). It is also compatible with `headerOnRow()` (previous rows will be skipped, and custom headers will be used instead of headers defined in the file).

It also affects the output of `getHeaders()` by returning only the custom headers you have defined, so the new `getOriginalHeaders()` method can be used to get the header row actually defined in the file, if required.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -107,6 +107,25 @@ $rows = SimpleExcelReader::create($pathToCsv)
...
 ```
 
+#### Manually setting the headers
+
+If you would like to use a specific array of values for the headers, you can use the `setHeaders()` method.
+
+```php
+// $rows is an instance of Illuminate\Support\LazyCollection
+$rows = SimpleExcelReader::create($pathToCsv)
+    ->setHeaders(['email_address', 'given_name'])
```
Comment (freekmurze) [inline README.md:117]: let's rename this to `useHeaders`, feels a tad more natural.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Looks good, could you rename that function? After that we're good to go. I appreciate the updated tests and readme 👍

File: README.md (hunk 1 above)
Comment (kitbs) [reply README.md]: Great, I have made this change.

**Comment by freekmurze** [comment]
Thank you!

**Pull request merged**